kong/router.lua

```diff
@@ -1533,76 +1673,118 @@ function _M.new(routes)
 
     -- header match
 
-    for _, header_name in ipairs(plain_indexes.headers) do
-      if req_headers[header_name] then
-        req_category = bor(req_category, MATCH_RULES.HEADER)
-        hits.header_name = header_name
-        break
+    local headers_key do
+      local headers_count = 0
+      if match_headers then
+        for i = 1, plain_indexes.headers[0] do
+          local name = plain_indexes.headers[i]
+          local value = req_headers[name]
+          if value then
+            if type(value) == "table" then
+              value = clone(value)
+              for i = 1, #value do
+                value[i] = lower(value[i])
+              end
+              sort(value)
+              value = concat(value, ", ")
+
+            else
```

Comment:
can `value` be a number? (for things like `Content-Length: 14`). If yes, then the `lower(value)` below will fail on that.

Reply:
I don't think so, and `string.lower` does not fail with numbers either:
```
resty -e 'print(string.lower(12))'                                                                                                          ✘ INT 11s
12
```